src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1278,6 +1257,67 @@ public void rebuild(String sourceDc, String keyspace)
         }
     }
 
+    public Set<String> getKeyspacesWithAllRangesAvailable(String sourceDc)
```

Comment:
This is a bit verbose so I'm open to suggestions. Figured I'd start out with the most literal name first though

Reply:
Probably fine.